### DISCUSSION TITLE
Non linearity when using hi intensity fields

### DISCUSSION THREAD

**Discussion by mojv** [NanoComp/meep#2140] [Q&A] [answered]
Hi, I'm using the gold material that is included in the `materials.py`, that as it says in the file, it allows one to use wave from .24797 μm to 6.1992 μm.  On the other hand,  meep allows also to create a source with a high spectral content, so if for example we create a pulse with spectral content of about 600 nm, that will translates to a temporal width of the order of 1 fs. 

As far as I understand, the default materials included in `materials.py` are linear, but a pulse of 1 fs would generate a great intensity, therefore the macroscopic polarization of the system will be nonlinear. does meep take that into account, or the only solutions is to no simulate a source a big spectral content.

**Comment by stevengj** [comment, marked as answer]
If you want nonlinear materials you need to set the `chi3` or `chi2` properties yourself.   Otherwise it defaults to linear materials (regardless of whether this is physical).

However, a typically reason to create an extremely narrow pulse in Meep is to Fourier transform the output to obtain the broadband response, i.e. when you really just want the *linear* frequency domain response over a broad bandwidth.  In this case, using linear materials is entirely appropriate.

Note also that whether "a pulse of 1 fs would generate a great intensity" depends also on the *amplitude* of the pulse.  On a computer (unlike in a real experiment), you can use as small an amplitude as you like (and with arbitrary units).

In general, for serious work, I would recommend defining your own materials rather than using the materials library.  That way you can use a minimal model (as few polarization fields as possible) to fit *your* gold parameters in your bandwidth of interest, and you can also include nonlinearities etc as desired.
